web/tests/reload.test.cjs

```diff
@@ -36,3 +40,32 @@ run_test("old_token_is_stale ", () => {
         ),
     );
 });
+
+run_test("reload", ({override}) => {
+    let idle_timeout_created = false;
+    let idle_timeout_canceled = false;
+    override(document, "to_$", () => ({
+        idle() {
+            idle_timeout_created = true;
+            return {
+                cancel() {
+                    idle_timeout_canceled = true;
+                },
+            };
+        },
+    }));
+    channel.get = (opts) => {
+        assert.equal(opts.url, "/compatibility");
+        opts.success();
+    };
+
+    reload.initiate({});
+    assert.equal(idle_timeout_created, true);
+    assert.equal(idle_timeout_canceled, false);
+
+    idle_timeout_created = false;
+
+    reload.maybe_reload_after_compose_end();
```

Comment:
Probably would be good to add a bit more test coverage here -- show that nothing happens for `maybe_reload_after_compose_start` and `maybe_reload_after_compose_end` if no reload has been initiated yet, and show that `maybe_reload_after_compose_start` also cancels and creates a new timeout.